arch/sim/src/Makefile

```diff
@@ -316,7 +316,8 @@ export_startup: board/libboard$(LIBEXT) up_head.o $(HOSTOBJS)
 depend: .depend
 
 cleanrel:
-	$(Q) rm -f nuttx.rel nuttx-names.dat
+	$(call DELFILE, nuttx-names.dat)
+	$(call DELFILE, nuttx.rel)
```

Comment:
Would this fix a race condition issue?
This change is okay by itself, but I don't see how it addresses a race condition. 
Isn't line 293 that can cause such a problem?
https://github.com/apache/incubator-nuttx/blob/d6a0da41d55d342aa5b434b98282c76cd996f1d3/arch/sim/src/Makefile#L293

Reply:
Race condition just in the order of deleting files, delete the nuttx.rel first will cause the target regeneration again, if the nuttx-names.dat deleted at this time, the compile will fail.

```
nuttx.rel : libarch$(LIBEXT) board/libboard$(LIBEXT) nuttx-names.dat $(LINKOBJS)
  $(Q) echo "LD:  nuttx.rel"
 /*     nuttx-names.dat will be deleted at this phase    */
  $(Q) $(LD) -r $(LDLINKFLAGS) $(RELPATHS) $(EXTRA_LIBPATHS) -o $@ $(REQUIREDOBJS) $(LDSTARTGROUP) $(RELLIBS) $(EXTRA_LIBS) $(LDENDGROUP) $(LDUNEXPORTSYMBOLS)
ifneq ($(HOSTOS),Darwin)
  $(Q) $(OBJCOPY) --redefine-syms=nuttx-names.dat $@  /* nuttx-names.dat has been deleted */
endif
```


 nuttx-names.dat to be delete later

Reply:
Okay.  Thanks for the explanation.